mapper-runtime/src/main/java/com/datastax/oss/driver/internal/mapper/MapperContext.java

```diff
@@ -65,4 +75,30 @@ public CqlIdentifier getKeyspaceId() {
   public CqlIdentifier getTableId() {
     return tableId;
   }
+
+  /**
+   * Returns an instance of the given converter class.
+   *
+   * <p>The results of this method are cached at the mapper level. If no instance of this class
+   * exists yet for this mapper, a new instance is built by looking for a public no-arg constructor.
+   */
+  public NameConverter getNameConverter(Class<? extends NameConverter> converterClass) {
+    return nameConverterCache.computeIfAbsent(converterClass, MapperContext::buildNameConverter);
+  }
+
+  private static NameConverter buildNameConverter(Class<? extends NameConverter> converterClass) {
```

Comment:
I think that code could be refactored to be more generic - `buildInstanceUsingNoArgsConstructor` and the reused in multiple places (I think that we are constructing newInstances() in this way in couple places in the code and we could replace that code we invocation of this method)

Reply:
I think that's the first place in the mapper where we use reflection. But it will be used again for [JAVA-2118](https://datastax-oss.atlassian.net/browse/JAVA-2118), I'll see if some reuse is possible.